tests/foreman/ui_airgun/test_reporttemplates.py

```diff
@@ -407,12 +191,104 @@ def test_positive_e2e_crud(session, module_org, module_loc):
+        assert session.reporttemplate.is_locked(new_name) is False
+        # CLONE
+        session.reporttemplate.clone(
+            new_name,
+            {
+                'template.name': clone_name,
+            }
+        )
+        crt = entities.ReportTemplate().search(
+                query={'search': 'name={0}'.format(clone_name)})[0].read()
+        assert crt.name == clone_name
+        assert crt.organization[0].read().name == module_org.name
+        assert crt.location[0].read().name == module_loc.name
+        assert crt.template == content
+        assert crt.default is True
+        # export
+        file_path = session.reporttemplate.export(new_name)
+        assert os.path.isfile(file_path)
+        with open(file_path) as dfile:
+            dfile_str = dfile.read()
+            assert content in dfile_str
+        # D
         session.reporttemplate.delete(new_name)
-        assert not session.reporttemplate.search(new_name)
+        assert len(entities.ReportTemplate().search(
+            query={'search': 'name={0}'.format(new_name)})) == 0
+
+
+@tier2
+def test_positive_generate_registered_hosts_report(session, module_org, module_loc):
+    """Use provided Registered Hosts report for testing
+
+    :id: b44d4cd8-a78e-47cf-9993-0bb871ac2c96
+
+    :expectedresults: All subtests pass
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+    """
+    # generate Host Status report
+    os_name = 'comma,' + gen_string('alpha')
+    os = entities.OperatingSystem(name=os_name).create().read()
+    host_template = entities.Host(organization=module_org,
+                                  location=module_loc, operatingsystem=os)
+    host_template.create_missing()
+    host_template2 = entities.Host(organization=module_org,
+                                   location=module_loc, operatingsystem=os)
+    host_template2.create_missing()
+    host_template3 = entities.Host(organization=module_org,
+                                   location=module_loc, operatingsystem=os)
+    host_template3.create_missing()
+    with session:
+        # create multiple hosts to test filtering
+        create_fake_host(
+            session,
+            host_template2,
+        )
+        host_name = create_fake_host(
+            session,
+            host_template,
+        )
+        create_fake_host(
+            session,
+            host_template3,
+        )
+        import pdb
+        pdb.set_trace()
+        file_path = session.reporttemplate.generate("Registered hosts",
+                                                    values={'inputs': {'Hosts filter': host_name}})
+        with open(file_path) as csvfile:
+            reader = csv.reader(csvfile)
+            columns = next(reader)
+            assert columns == ['Name', 'Ip', 'Operating System',
+                               'Subscriptions', 'Applicable Errata', 'Owner',
+                               'Kernel', 'Latest kernel available']
```

Comment:
I'm not 100% sure, but I think you can start with `csv.DictReader()`, assert number of items in list and then, for first item, assert that this dict has each of these column names. That would save `csvfile.seek()` call and duplicated invocation of `csv.reader()`.